homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -252,9 +288,26 @@ async def _async_handle_message(
             "gemini-1.0" not in model_name and "gemini-pro" not in model_name
         )
 
-        prompt, *messages = [
-            _chat_message_convert(message) for message in session.async_get_messages()
-        ]
+        prompt = chat_log.content[0].content  # type: ignore[union-attr]
+        messages: list[genai_types.ContentDict] = []
+
+        # Google groups tool results, we do not. Group them before sending.
+        tool_results: list[conversation.ToolResultContent] = []
+
+        for chat_content in chat_log.content[1:]:
+            if chat_content.role == "tool_result":
+                tool_results.append(chat_content)  # type: ignore[arg-type]
+
+            elif tool_results:
```

Comment:
Will this skip a `chat_content` after a `tool_result`? I think this may nee dto be a check inside the `else`.

Reply:
Good catch.